filters/auth/oidc.go

```diff
@@ -327,6 +328,25 @@ func (f *tokenOidcFilter) doOauthRedirect(ctx filters.FilterContext) {
 		return
 	}
 
+	for i, param := range f.authCodeOptions {
+		r := reflect.ValueOf(param)
```

Comment:
May I suggest a way out of reflection?

We may have two slices `f.authCodeOptions` and `f.queryParams` and move  `if v == "skipper-request-query"` into `CreateFilter` so that here we create new slice, copy  `f.authCodeOptions`  into it and for each `p` in  `f.queryParams` we append new `oauth2.SetAuthURLParam` to this new slice...

Reply:
Of course `if f.queryParams) == nil` we can skip the copy altogether.

Reply:
That is a very good suggestion. I changed the code. Is that what you had in mind? Thank you @AlexanderYastrebov !

Reply:
Yes, I have posted some comments below.